src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -1961,9 +1960,31 @@ void Session::send_bind_message()
     int protocol_version = m_conn.get_negotiated_protocol_version();
     OutputBuffer& out = m_conn.get_output_buffer();
     // Discard the token since it's ignored by the server.
-    std::string empty_access_token{};
-    protocol.make_bind_message(protocol_version, out, session_ident, path, empty_access_token, need_client_file_ident,
-                               is_subserver); // Throws
+    std::string empty_access_token;
+    if (m_is_flx_sync_session) {
+        nlohmann::json bind_json_data;
+        if (auto migrated_partition = get_migration_store()->get_migrated_partition()) {
+            bind_json_data["migratedPartition"] = *migrated_partition;
+        }
+        if (logger.would_log(util::Logger::Level::debug)) {
+            std::string json_data_dump;
+            if (!bind_json_data.empty()) {
+                json_data_dump = bind_json_data.dump();
+            }
+            logger.debug(
+                "Sending: BIND(session_ident=%1, need_client_file_ident=%2 is_subserver=%3 json_data=\"%4\")",
+                session_ident, need_client_file_ident, is_subserver, json_data_dump);
+        }
+        protocol.make_flx_bind_message(protocol_version, out, session_ident, bind_json_data, empty_access_token,
+                                       need_client_file_ident, is_subserver); // Throws
+    }
+    else {
+        std::string server_path = get_virt_path();
+        logger.debug("Sending: BIND(session_ident=%1, need_client_file_ident=%2 is_subserver=%3 server_path=%4)",
+                     session_ident, need_client_file_ident, is_subserver, server_path);
+        protocol.make_pbs_bind_message(protocol_version, out, session_ident, server_path, empty_access_token,
```

Comment:
unrelated: is there a plan, at some point, to make pbs work with a json so we don't need to treat it differently than flx?

Reply:
I don't believe so - the JSON data was "shimmed" into the BIND message for FLX, since the existing server_path section, which is used to send the partition value in PBS, was not used by FLX. In addition, since we want to move away from PBS eventually, there likely won't be a need for adding JSON values.

Reply:
great!